File: c++/leetcode/11. Container With Most Water.cpp
class Solution
{
public:
  int maxArea(vector<int> &height)
  {
    int _max{};

    int start = 0;
    int end = height.size() - 1;

    while (start < end)
    {
      _max = max(_max, min(height[start], height[end]) * (end - start));

      if (height[start] < height[end])
        ++start;
      else
        --end;
    }

    return _max;
  }
};

File: javascript/leetcode/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let _max = 0,
    start = 0,
    end = height.length - 1;

  while (start < end) {
    _max = Math.max(_max, Math.min(height[start], height[end]) * (end - start));

    if (height[start] < height[end]) ++start;
    else --end;
  }

  return _max;
};
